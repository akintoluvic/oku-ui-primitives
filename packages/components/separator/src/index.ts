export {
  OkuLabel,
  label,
} from './separator'
export type { LabelProps } from './label'
