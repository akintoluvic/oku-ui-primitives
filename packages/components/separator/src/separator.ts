import { defineComponent, h, ref } from 'vue'
import type { ComponentPropsWithoutRef, ElementRef } from '@oku-ui/primitive'
import { Primitive } from '@oku-ui/primitive'

type PrimitiveSeparatorProps = ComponentPropsWithoutRef<typeof Primitive.div>

// interface SeparatorProps extends PrimitiveSeparatorProps {}
// type SeparatorElement = ElementRef<typeof Primitive.div>

const NAME = 'Separator'
const DEFAULT_ORIENTATION = 'horizontal'
const ORIENTATIONS = ['horizontal', 'vertical'] as const

type Orientation = typeof ORIENTATIONS[number]
type SeparatorElement = ElementRef<typeof Primitive.div>
interface SeparatorProps extends PrimitiveSeparatorProps {
  /**
   * Either `vertical` or `horizontal`. Defaults to `horizontal`.
   */
  orientation?: Orientation
  /**
   * Whether or not the component is purely decorative. When true, accessibility-related attributes
   * are updated so that that the rendered element is removed from the accessibility tree.
   */
  decorative?: boolean
}

const Separator = defineComponent<SeparatorProps>({
  name: NAME,
  inheritAttrs: false,
  setup(props, { attrs, slots, expose }) {
    const inferRef = ref<SeparatorElement>()

    expose({
      inferRef,
    })

    return () =>
      h(
        Primitive.div,
        {
          ...attrs,
          ref: inferRef,
        },
        slots.default?.(),
      )
  },
})

const OkuSeparator = Separator

export { OkuSeparator, Separator }
export type { SeparatorProps }
