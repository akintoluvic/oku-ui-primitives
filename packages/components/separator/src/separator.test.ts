import { describe, expect, it } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import { OkuSeparator } from './separator'

describe('OkuSeparator', () => {
  const wrapper = shallowMount(OkuSeparator)

  it('renders correctly', async () => {
    expect(wrapper).toMatchSnapshot()
  })
})
