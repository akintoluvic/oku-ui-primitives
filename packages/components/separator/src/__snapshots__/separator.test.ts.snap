// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OkuSeparator > renders correctly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <primitivediv-stub
        dataorientation="horizontal"
        role="separator"
      />
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {
          "innerRef": <primitivediv-stub
            dataorientation="horizontal"
            role="separator"
          />,
        },
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {
            "innerRef": <primitivediv-stub
              dataorientation="horizontal"
              role="separator"
            />,
          },
          "_value": {
            "innerRef": <primitivediv-stub
              dataorientation="horizontal"
              role="separator"
            />,
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {
          "innerRef": <primitivediv-stub
            dataorientation="horizontal"
            role="separator"
          />,
        },
      },
      "refs": {
        "VTU_COMPONENT": {
          "innerRef": <primitivediv-stub
            dataorientation="horizontal"
            role="separator"
          />,
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {
          "innerRef": <primitivediv-stub
            dataorientation="horizontal"
            role="separator"
          />,
        },
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "decorative": false,
            "orientation": "horizontal",
          },
          "da": null,
          "data": {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "innerRef": <primitivediv-stub
              dataorientation="horizontal"
              role="separator"
            />,
          },
          "exposed": {
            "innerRef": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": <primitivediv-stub
                dataorientation="horizontal"
                role="separator"
              />,
              "dep": undefined,
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
          },
          "inheritAttrs": false,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "decorative": false,
            "orientation": "horizontal",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "decorative": {
                "0": true,
                "1": true,
                "default": false,
                "type": [Function],
              },
              "orientation": {
                "0": false,
                "1": true,
                "default": "horizontal",
                "type": [Function],
              },
            },
            [
              "decorative",
              "orientation",
            ],
          ],
          "provides": {},
          "proxy": {
            "decorative": false,
            "orientation": "horizontal",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": <primitivediv-stub
                    dataorientation="horizontal"
                    role="separator"
                  />,
                  "dep": undefined,
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": null,
            "component": {
              "a": null,
              "accessCache": {
                "__v_isReadonly": 0,
                "__v_isShallow": 0,
                "__v_raw": 0,
                "__v_skip": 4,
              },
              "appContext": {
                "app": [Circular],
                "components": {
                  "transition": {
                    "name": "transition",
                  },
                  "transition-group": {
                    "name": "transition-group",
                  },
                },
                "config": {
                  "compilerOptions": {},
                  "errorHandler": undefined,
                  "globalProperties": {},
                  "isNativeTag": [Function],
                  "optionMergeStrategies": {},
                  "performance": false,
                  "warnHandler": undefined,
                },
                "directives": {},
                "emitsCache": WeakMap {},
                "mixins": [
                  {
                    "beforeCreate": [Function],
                  },
                ],
                "optionsCache": WeakMap {},
                "propsCache": WeakMap {},
                "provides": {},
                "reload": [Function],
              },
              "asyncDep": null,
              "asyncResolved": false,
              "attrs": {
                "aria-orientation": undefined,
                "dataOrientation": "horizontal",
                "role": "separator",
                "style": {
                  "border": "none",
                },
              },
              "bc": null,
              "bm": null,
              "bu": null,
              "bum": null,
              "c": null,
              "components": null,
              "ctx": {},
              "da": null,
              "data": {},
              "directives": null,
              "ec": null,
              "effect": ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
              "emit": [Function],
              "emitsOptions": null,
              "emitted": null,
              "exposeProxy": null,
              "exposed": null,
              "inheritAttrs": undefined,
              "isDeactivated": false,
              "isMounted": true,
              "isUnmounted": false,
              "m": null,
              "next": null,
              "parent": [Circular],
              "props": {},
              "propsDefaults": {},
              "propsOptions": [
                {},
                [],
              ],
              "provides": {},
              "proxy": {},
              "refs": {},
              "render": [Function],
              "renderCache": [],
              "root": [Circular],
              "rtc": null,
              "rtg": null,
              "scope": EffectScope {
                "_active": true,
                "cleanups": [],
                "detached": true,
                "effects": [
                  ReactiveEffect {
                    "active": true,
                    "allowRecurse": true,
                    "deps": [
                      Set {
                        [Circular],
                      },
                    ],
                    "fn": [Function],
                    "onTrack": undefined,
                    "onTrigger": undefined,
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                ],
                "parent": undefined,
              },
              "setupContext": {
                "attrs": {
                  "aria-orientation": undefined,
                  "dataOrientation": "horizontal",
                  "role": "separator",
                  "style": {
                    "border": "none",
                  },
                },
                "emit": [Function],
                "expose": [Function],
                "slots": {},
              },
              "setupState": {},
              "slots": {},
              "sp": null,
              "subTree": {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "ce": undefined,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <primitivediv-stub
                  dataorientation="horizontal"
                  role="separator"
                />,
                "key": null,
                "patchFlag": 16,
                "props": {
                  "aria-orientation": undefined,
                  "dataOrientation": "horizontal",
                  "role": "separator",
                  "style": {
                    "border": "none",
                  },
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "primitivediv-stub",
              },
              "suspense": null,
              "suspenseId": 0,
              "type": {
                "model": undefined,
                "name": "Primitivediv",
                "props": {},
                "setup": [Function],
              },
              "u": null,
              "uid": 2,
              "um": null,
              "update": [Function],
              "vnode": [Circular],
              "withProxy": null,
            },
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": null,
            "dynamicProps": null,
            "el": <primitivediv-stub
              dataorientation="horizontal"
              role="separator"
            />,
            "key": null,
            "patchFlag": 0,
            "props": {
              "aria-orientation": undefined,
              "dataOrientation": "horizontal",
              "ref": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": Set {
                  ReactiveEffect {
                    "active": true,
                    "computed": ComputedRefImpl {
                      "__v_isReadonly": true,
                      "__v_isRef": true,
                      "_cacheable": true,
                      "_dirty": false,
                      "_setter": [Function],
                      "_value": <primitivediv-stub
                        dataorientation="horizontal"
                        role="separator"
                      />,
                      "dep": undefined,
                      "effect": [Circular],
                    },
                    "deps": [
                      [Circular],
                    ],
                    "fn": [Function],
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                },
              },
              "role": "separator",
              "style": {
                "border": "none",
              },
            },
            "ref": {
              "f": false,
              "i": [Circular],
              "k": undefined,
              "r": RefImpl {
                "__v_isRef": true,
                "__v_isShallow": false,
                "_rawValue": {},
                "_value": {},
                "dep": Set {
                  ReactiveEffect {
                    "active": true,
                    "computed": ComputedRefImpl {
                      "__v_isReadonly": true,
                      "__v_isRef": true,
                      "_cacheable": true,
                      "_dirty": false,
                      "_setter": [Function],
                      "_value": <primitivediv-stub
                        dataorientation="horizontal"
                        role="separator"
                      />,
                      "dep": undefined,
                      "effect": [Circular],
                    },
                    "deps": [
                      [Circular],
                    ],
                    "fn": [Function],
                    "parent": undefined,
                    "scheduler": [Function],
                  },
                },
              },
            },
            "scopeId": null,
            "shapeFlag": 4,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": {
              "model": undefined,
              "name": "Primitivediv",
              "props": {},
              "setup": [Function],
            },
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "components": {},
            "inheritAttrs": false,
            "name": "Separator",
            "props": {
              "decorative": {
                "default": false,
                "type": [Function],
              },
              "orientation": {
                "default": "horizontal",
                "type": [Function],
              },
            },
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <primitivediv-stub
          dataorientation="horizontal"
          role="separator"
        />,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "components": {},
          "inheritAttrs": false,
          "name": "Separator",
          "props": {
            "decorative": {
              "default": false,
              "type": [Function],
            },
            "orientation": {
              "default": "horizontal",
              "type": [Function],
            },
          },
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <primitivediv-stub
          dataorientation="horizontal"
          role="separator"
        />,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.47",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {
    "innerRef": <primitivediv-stub
      dataorientation="horizontal"
      role="separator"
    />,
  },
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {
      "innerRef": <primitivediv-stub
        dataorientation="horizontal"
        role="separator"
      />,
    },
  },
  "wrapperElement": <primitivediv-stub
    dataorientation="horizontal"
    role="separator"
  />,
}
`;
